File: UI/trackit_admin/lib/models/Subscription/subscription.dart
import 'package:json_annotation/json_annotation.dart';
part 'subscription.g.dart';

@JsonSerializable()
class Subscription {
  int? subscriptionId;
  int? generalUserId;
  DateTime? purchaseDate;

  Subscription(this.subscriptionId, this.generalUserId, this.purchaseDate);

  factory Subscription.fromJson(Map<String, dynamic> json) =>
      _$SubscriptionFromJson(json);

  Map<String, dynamic> toJson() => _$SubscriptionToJson(this);
}

File: TrackIt.Services/Database/Subscription.cs
namespace TrackIt.Services.Database;

public partial class Subscription
{
	public int SubscriptionId { get; set; }

	public int GeneralUserId { get; set; }

	public DateTime PurchaseDate { get; set; }

	public virtual GeneralUser GeneralUser { get; set; } = null!;
}
